projects/rocprofiler-systems/source/lib/rocprof-sys/library/amd_smi.cpp

```diff
@@ -121,34 +120,35 @@ rocpd_initialize_smi_pmc(size_t gpu_id)
     const char* EXPRESSION       = "";
     const char* CELSIUS_DEGREES  = "\u00B0C";
     auto        ni               = node_info::get_instance();
-    const auto* TARGET_ARCH      = "GPU";
-
-    auto& _agent_manager = agent_manager::get_instance();
-    auto  base_id = _agent_manager.get_agent_by_id(gpu_id, agent_type::GPU).base_id;
-
-    data_processor.insert_pmc_description(
-        ni.id, getpid(), base_id, TARGET_ARCH, EVENT_CODE, INSTANCE_ID,
-        trait::name<category::amd_smi_mm_busy>::value, "Busy",
-        trait::name<category::amd_smi_mm_busy>::description, LONG_DESCRIPTION, COMPONENT,
-        "%", "ABS", BLOCK, EXPRESSION, 0, 0);
-
-    data_processor.insert_pmc_description(
-        ni.id, getpid(), base_id, TARGET_ARCH, EVENT_CODE, INSTANCE_ID,
-        trait::name<category::amd_smi_temp>::value, "Temp",
-        trait::name<category::amd_smi_temp>::description, LONG_DESCRIPTION, COMPONENT,
-        CELSIUS_DEGREES, "ABS", BLOCK, EXPRESSION, 0, 0);
-
-    data_processor.insert_pmc_description(
-        ni.id, getpid(), base_id, TARGET_ARCH, EVENT_CODE, INSTANCE_ID,
-        trait::name<category::amd_smi_power>::value, "Pow",
-        trait::name<category::amd_smi_power>::description, LONG_DESCRIPTION, COMPONENT,
-        "w", "ABS", BLOCK, EXPRESSION, 0, 0);
-
-    data_processor.insert_pmc_description(
-        ni.id, getpid(), base_id, TARGET_ARCH, EVENT_CODE, INSTANCE_ID,
-        trait::name<category::amd_smi_memory_usage>::value, "MemUsg",
-        trait::name<category::amd_smi_memory_usage>::description, LONG_DESCRIPTION,
-        COMPONENT, "MB", "ABS", BLOCK, EXPRESSION, 0, 0);
+    const char* TARGET_ARCH      = "GPU";
+
+    auto& agent_mngr = agent_manager::get_instance();
+    auto  agent_handle =
+        agent_mngr.get_agent_by_type_index(gpu_id, agent_type::GPU).handle;
+
+    trace_cache::get_metadata_registry().add_pmc_info(
+        { agent_handle, TARGET_ARCH, EVENT_CODE, INSTANCE_ID,
+          trait::name<category::amd_smi_mm_busy>::value, "Busy",
+          trait::name<category::amd_smi_mm_busy>::description, LONG_DESCRIPTION,
+          COMPONENT, "%", "ABS", BLOCK, EXPRESSION, 0, 0, "{}" });
+
+    trace_cache::get_metadata_registry().add_pmc_info(
+        { agent_handle, TARGET_ARCH, EVENT_CODE, INSTANCE_ID,
+          trait::name<category::amd_smi_temp>::value, "Temp",
+          trait::name<category::amd_smi_temp>::description, LONG_DESCRIPTION, COMPONENT,
+          CELSIUS_DEGREES, "ABS", BLOCK, EXPRESSION, 0, 0 });
+
+    trace_cache::get_metadata_registry().add_pmc_info(
+        { agent_handle, TARGET_ARCH, EVENT_CODE, INSTANCE_ID,
+          trait::name<category::amd_smi_power>::value, "Pow",
+          trait::name<category::amd_smi_power>::description, LONG_DESCRIPTION, COMPONENT,
+          "w", "ABS", BLOCK, EXPRESSION, 0, 0 });
```

Comment:
```suggestion
          "W", "ABS", BLOCK, EXPRESSION, 0, 0 });
```
Capitalized "W" for Watts.